.script/tests/detectionTemplateSchemaValidation/DetectionTemplateSchemaValidationTests.cs

```diff
@@ -10,19 +12,50 @@
 
 namespace Kqlvalidations.Tests
 {
-    public class DetectionTemplateStructureValidationTests
+    public class DetectionTemplateSchemaValidationTests
     {
         private static readonly string DetectionPath = DetectionsYamlFilesTestData.GetDetectionPath();
 
         [Theory]
         [ClassData(typeof(DetectionsYamlFilesTestData))]
         public void Validate_DetectionTemplates_HaveValidTemplateStructure(string detectionsYamlFileName)
+        {
+            Action<ScheduledTemplateInternalModel> validateConnectorSchema = (ScheduledTemplateInternalModel templateObject) => {
+                var validationContext = new ValidationContext(templateObject);
+                Validator.ValidateObject(templateObject, validationContext, true);
+            };
+
+            ValidateDetectionTemplate(detectionsYamlFileName, validateConnectorSchema);
+        }
+
+        [Theory]
+        [ClassData(typeof(DetectionsYamlFilesTestData))]
+        public void Validate_DetectionTemplates_HaveValidConnectorsIds(string detectionsYamlFileName)
+        {
+            Action<ScheduledTemplateInternalModel> validateConnectorsIds = (ScheduledTemplateInternalModel templateObject) => {
+                List<string> connectorIds = templateObject.RequiredDataConnectors
+                    .Where(requiredDataConnector => requiredDataConnector.ConnectorId != null)
+                    .Select(requiredDataConnector => requiredDataConnector.ConnectorId).ToList();
+
+                connectorIds.ForEach(connectorId => {
```

Comment:
nit - you can do join of the two lists and see the size of the join is the same as the size of the RequiredDataConnectors in the template

Reply:
I dont understand what you meant.
I need to verify the content of connectorId inside each requiredDataConnectors, how checking the size will help me ?
@oshvartz